src/cata_utility.cpp

```diff
@@ -46,15 +46,16 @@ bool isBetween( int test, int down, int up )
 
 bool lcmatch( const std::string &str, const std::string &qry )
 {
-    std::string needle;
-    needle.reserve( qry.size() );
-    std::transform( qry.begin(), qry.end(), std::back_inserter( needle ), tolower );
+    std::locale::global( std::locale( "ru_RU.utf8" ) );
```

Comment:
This creates a new `locale` (with the given name) and sets it as the global `locale`.

In the line below, you create a default constructed `locale`, which will be a copy of the global `locale` (set above).

I assume you can use `std::locale( "ru_RU.utf8" )` instead of a default constructed object and don't need to change the global `locale`. But why use a named locale anyway? The global locale should already be the correct one (the one set in the options), or is the global locale not set by the program at all?

Reply:
Is seems `setlocale( LC_ALL, NULL )` always returns "ru_RU.UTF-8" for me no matter what language is set in the options.

You are right, that line not needed in case we get proper locale at the moment we call `use_facet`.
This form:
```
auto &f = std::use_facet<std::ctype<wchar_t>>( std::locale() );
```
But this
```
auto &f = std::use_facet<std::ctype<wchar_t>>( std::locale( "ru_RU.utf8" ) );
```
causes `SIGSEGV: Segmentation fault` on line `f.tolower( &whaystack[0], &whaystack[0] + whaystack.size() );`